codebase/codebase.go

```diff
@@ -37,22 +40,41 @@ func (c *CodebaseContent) ToMap() map[string]interface{} {
 	res[BranchKey] = c.Branch
 	res[FileNameKey] = c.FileName
 	res[LineNumberKey] = c.LineNumber
+	res[CodebaseIDKey] = c.CodebaseID
 	return res
 }
 
+// validateURL makes sure Repo is valid GIT URL
+func validateURL(repo string) (string, bool) {
+	validPrefixes := []string{"https://github.com/", "git@github.com:"}
```

Comment:
For validation, maybe we could use this regex pattern `(?:git|ssh|https?|git@[-\w.]+):(\/\/)?(.*?)(\.git)(\/?|\#[-\d\w._]+?)$`